### ISSUE TITLE
Migrating from JUnit 4/3 to JUnit 5 for testing

### ISSUE THREAD

**Issue by krisstern** [jenkinsci/build-timeout-plugin#85]
### What feature do you want to see added?

Currently the codebase uses JUnit 4 for testing. Would be ideal if the tests could be updated to use JUnit 5 for this plugin, in order to harness the latest features from Java 8 or later, such as the lambda functions, rendering the tests more powerful and easier to maintain. 

More info about the major differences and benefits of migrating from JUnit 4 to JUnit 5 can be found for example in [this blog post](https://blogs.oracle.com/javamagazine/post/migrating-from-junit-4-to-junit-5-important-differences-and-benefits) from Oracle.

**Comment by baruKreddy** [comment]
@krisstern I would like to work on this. is there any jira task created for this issue?

**Comment by krisstern** [comment]
Hi @baruKreddy! We are not currently using JIRA. You are welcome to start work on the issue.

**Comment by baruKreddy** [comment]
Thanks:) @krisstern

**Comment by baruKreddy** [comment]
@krisstern I can see https://github.com/jenkinsci/junit-plugin/blob/master/pom.xml the version is managed in parent still. So where should we go for a version change.
 <parent>
        <groupId>org.jenkins-ci.plugins</groupId>
        <artifactId>plugin</artifactId>
        <version>4.31</version>
        <relativePath />
    </parent>

**Comment by krisstern** [comment]
The issue now is split into two parts:
1. Non-integration tests
2. Integration tests

**Comment by Urjaswi-2655** [comment]
Hi, @krisstern ,I would like to work on Migrating from JUnit 4 to JUnit 5 for Integration testing issue .
I have no prior experience in testing, but I would like to learn.
Pls help me to get started. 
Before asking for help,I read this doc to understand Integration Testing ( https://www.javatpoint.com/integration-testing )
And read the document for major differences and benefits of migrating from JUnit 4 to JUnit 5(Although , I have not understood everything).
Should I have to learn about about JUnit 5 , and Is this course suitable to understand it (https://www.udemy.com/course/mockito-tutorial-with-junit-examples/ )  ?
Please help,if possibe,Thanks.

**Comment by krisstern** [comment]
There is this book called ["Junit in Action", 3rd edition](https://www.manning.com/books/junit-in-action-third-edition) that may be useful for this task. You can find free PDF copies on the internet or pay for a copy. They do have a chapter dealing migrating from Junit 4 to Junit 5 in particular. (I have access to it via my O'Reilly subscription.) 

I think the book would be a good starter to begin on a PR to complete this issue.

**Comment by Urjaswi-2655** [comment]
Thanks, @krisstern I downloaded that book . (I will read it)
Just one last request  Pls can you verify this course (https://www.udemy.com/course/mockito-tutorial-with-junit-examples/)
for me or prescribe some other video related course because I want to contribute to GSOC 22 , and there is similar kind of testing idea present in one organizations GSOC ideas page. And I think that at first going through a course will help me easily navigate through this book , otherwise it will be time taking process, definitely I will go through the preferred book. 
Thank You for replying.

**Comment by Urjaswi-2655** [comment]
Thanks, @krisstern I downloaded that book . (I will read it)

**Comment by krisstern** [comment]
HI @Urjaswi-2655 If you are looking for relevant Udemy courses there is a free one https://www.udemy.com/course/junit-quick-start-for-beginners-java-unit-testing/. But this does not dive deep into migration from Junit 4 to Junit 5. So naturally you should fall back on the book preferably.

**Comment by i-laird** [comment]
@Urjaswi-2655 How is this going?

**Comment by shamlymhd** [comment]
Hi @krisstern I would like to work on this issue. can I start work on this?

**Comment by krisstern** [comment]
Hi @shamlymhd Sure, I don't think anyone else is working on the issue at the moment

**Comment by priyadarshan85** [comment]
Is anyone working on this issue?

**Comment by krisstern** [comment]
No, don't think so

**Comment by priyadarshan85** [comment]
I see that the migration to Junit 5 is already done. Whats pending on this issue?

**Comment by krisstern** [comment]
It's not 100% completed yet, only partially done.

**Comment by priyadarshan85** [comment]
All test cases are not passing ? Is it? Because pom.xml shows the upgrade to latest junit 5 version

**Comment by priyadarshan85** [comment]
@krisstern Getting following error when I build the project:

<img width="1407" alt="image" src="https://user-images.githubusercontent.com/100334324/194693571-85c78545-69dc-4833-b515-8ff1016e6c11.png">


Am I missing something here

**Comment by krisstern** [comment]
> All test cases are not passing ? Is it? Because pom.xml shows the upgrade to latest junit 5 version. Running tests now

I think it is the codebase that needs upgrading, as supposedly the "integration" tests have not been updated to JUnit 5 according to some comments of a previoius contributor. The tests are all passing. I can build it locally with no issues. 

<img width="851" alt="Screenshot 2022-10-08 at 2 59 02 PM" src="https://user-images.githubusercontent.com/88480540/194694473-01add3b7-27de-43b5-b154-077fc57f2692.png">
